core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQProducer.scala

```diff
@@ -39,9 +40,8 @@ class DefaultRabbitMQProducer[F[_]: FromTask, A: ProductConverter](name: String,
 
   override def send(routingKey: String, body: A, properties: Option[MessageProperties] = None): F[Unit] = {
     val finalProperties = {
-      val messageProperties = converter.fillProperties {
-        properties.getOrElse(MessageProperties(messageId = Some(UUID.randomUUID().toString)))
-      }
+      val initialProperties = properties.getOrElse(defaultProperties.copy(messageId = Some(UUID.randomUUID().toString)))
```

Comment:
Didn't we talk about _merging_? This seems like A or B...

Reply:
Yep, it was the original idea but it's not so easy in fact - the problem is meaning of `None` value.

Does it mean that you want to keep the default value, or do you want to overwrite the default value with _empty_ value? So I decided to use this approach. It's also easier to understand IMHO.

Reply:
That's a hard decision indeed. What do think @rpliva and @petr-k?

Reply:
I'm on the fence about this one.. This behavior is simpler so I'd stick to it, but needs to be stressed in the docs, because I imagine it might be somehow unexpected to many.

Reply:
OK, so please @augi reflect it in the docs and it will be ready to merge.